engine/src/org/pentaho/di/trans/steps/exceloutput/ExcelOutput.java

```diff
@@ -681,7 +681,7 @@ private boolean closeFile() {
         }
 
       }
-      // data.formats.clear();
+      data.formats.clear();
```

Comment:
Important note - that line was commented out when the "Append in Excel Output" feature was created - SVN revision 6294 on 12/31/2007. So when testing, you have to test all aspects of this step including appending.